tasks/connectors/qualys_was/qualys_was.rb

```diff
@@ -230,7 +235,12 @@ def domain_detail(find_from)
       end
 
       def name(find_from)
-        "#{find_from['qid']}-#{find_from['name']}"
+        if @options[:match_finding_with_vuln_def]
+          structured_name = [find_from['qid'], find_from['id'], find_from['name']].compact.join('-')
+          structured_name unless structured_name.empty?
```

Comment:
In the odd case that structured_name was empty, this would result in returning nil for finding scanner_id and vuln_def name. I still think that if we are checking for nil/empty values in toolkit that we need to make sure we are not ingesting nil values into Kenna.

Reply:
I agree with that. However, doing the  nil value check in a separate pull request could make more sense? Since this pull request is addressing the issue on mismatch between svd and findings.

Reply:
Added the safeguard check in the code